File: app/src/main/java/com/sxshi/retrofit/ui/main/MainFragment.java
package com.sxshi.retrofit.ui.main;

import android.view.View;

import com.sxshi.retrofit.R;
import com.sxshi.retrofit.base.BaseFragment;

import java.util.HashMap;

public class MainFragment extends BaseFragment<MainViewModel> {
    public static MainFragment newInstance() {
        return new MainFragment();
    }

    @Override
    protected void initData() {
        model.getUserInfoLiveData().observe(this, s -> {
            // TODO: 2019-12-12  update UI

        });
    }

    @Override
    protected void initView(View view) {
        view.findViewById(R.id.start).setOnClickListener(v -> {
            HashMap<String, String> param = new HashMap<>();
            model.Login(param);
        });
    }

    @Override
    protected Class<MainViewModel> providerVM() {
        return MainViewModel.class;
    }

    @Override
    protected int getContentLayoutResId() {
        return R.layout.main_fragment;
    }

}

File: app/src/main/java/com/sxshi/retrofit/ui/main/MainFragment.kt
package com.sxshi.retrofit.ui.main

import android.view.View
import androidx.lifecycle.Observer
import com.sxshi.retrofit.R
import com.sxshi.retrofit.base.BaseFragment
import java.util.*

class MainFragment : BaseFragment<MainViewModel>() {
    override fun initData() {
        model?.userInfoLiveData?.observe(this, Observer { s: String? -> })
    }

    override fun initView(view: View) {
        view.findViewById<View>(R.id.start).setOnClickListener { v: View ->
            val param = HashMap<String, String>()
            model?.login(param)
        }
    }

    override fun providerVM(): Class<MainViewModel>? {
        return MainViewModel::class.java
    }


    override val contentLayoutResId: Int = R.layout.main_fragment

    companion object {
        fun newInstance(): MainFragment {
            return MainFragment()
        }
    }
}